kadas/app/milx/kadasmilxintegration.cpp

```diff
@@ -412,6 +390,168 @@ void KadasMilxIntegration::saveMilxly()
   delete zip;
 }
 
+void KadasMilxIntegration::exportKml()
+{
+  kApp->unsetMapTool();
+
+  KadasMilxExportDialog exportDialog;
+  exportDialog.setExportMode( KadasMilxExportDialog::ExportMode::Kml );
+
+  if ( exportDialog.exec() == QDialog::Rejected )
+  {
+    return;
+  }
+
+  QStringList exportLayers = exportDialog.selectedLayers();
+
+  QStringList filters;
+  filters.append( tr( "Compressed KML Layer (*.kmz)" ) );
+  filters.append( tr( "KML Layer (*.kml)" ) );
+
+  QString lastDir = QgsSettings().value( "/UI/lastImportExportDir", "." ).toString();
+  QString selectedFilter;
+
+  QString filename = QFileDialog::getSaveFileName( 0, tr( "Select Output" ), lastDir, filters.join( ";;" ), &selectedFilter );
+  if ( filename.isEmpty() )
+  {
+    return;
+  }
+  QgsSettings().setValue( "/UI/lastImportExportDir", QFileInfo( filename ).absolutePath() );
+  if ( selectedFilter == filters[0] && !filename.endsWith( ".kmz", Qt::CaseInsensitive ) )
+  {
+    filename += ".kmz";
+  }
+  else if ( selectedFilter == filters[1] && !filename.endsWith( ".kml", Qt::CaseInsensitive ) )
+  {
+    filename += ".kml";
+  }
+  QString currentVersionTag; KadasMilxClient::getCurrentLibraryVersionTag( currentVersionTag );
+
+  QIODevice *dev = 0;
+  QuaZip *zip = 0;
```

Comment:
```suggestion
  std::unique_ptr<QIODevice> dev;
  std::unique_ptr<QuaZip> zip;
```

could we get rid of manual memory management here?